src/wallet/components/WalletDropdownBaseName.tsx

```diff
@@ -0,0 +1,58 @@
+import { base } from 'viem/chains';
+import { useAccount } from 'wagmi';
+import { useName } from '../../identity/hooks/useName';
+import { Spinner } from '../../internal/components/Spinner';
+import { baseNameSvg } from '../../internal/svg/baseNameSvg';
+import { cn, pressable, text } from '../../styles/theme';
+import type { WalletDropdownBaseNameReact } from '../types';
+
+export function WalletDropdownBaseName({
+  className,
+}: WalletDropdownBaseNameReact) {
+  const account = useAccount();
+
+  const { data: baseName, isLoading } = useName({
+    address: account.address as `0x${string}`,
+    chain: base,
+  });
+
+  let hasBaseUserName = false;
+
+  if (baseName) {
+    hasBaseUserName = true;
+  }
+
+  const title = hasBaseUserName ? 'Go to profile' : 'Claim a Basename';
+  const href = hasBaseUserName
+    ? `https://www.base.org/name/${baseName}`
+    : 'https://www.base.org/names';
+
+  return (
+    <a
+      className={cn(
+        pressable.default,
+        'flex items-center px-4 py-2',
+        className,
+      )}
+      href={href}
+      target="_blank"
+      rel="noopener noreferrer"
+    >
+      {baseNameSvg && <div className="mr-2 h-5 w-5">{baseNameSvg}</div>}
+      <div className="flex items-center">
+        {isLoading ? (
+          <Spinner />
+        ) : (
+          <>
+            <span className={cn(text.body, 'shrink-0')}>{title}</span>
+            {!hasBaseUserName && (
+              <span className="ml-2 rounded-full bg-[#E0E7FF] px-2 text-center font-bold font-inter text-[#4F46E5] text-[0.6875rem] uppercase leading-6">
```

Comment:
Let's use the `cn` utils when we have very long CSS stings, it helps with reviewing.

Reply:
Do we want to use `cn` here though? This is a specific use case, customizing the `NEW` text.

Reply:
Wrapped in `cn`

Reply:
I tried to apply existing CN templates for styling. It's a little tricky because this text is so different. Therefore I wrapped the classes in cn but am not using any default styling.